mirage/awa_mirage.ml

```diff
@@ -355,6 +388,18 @@ module Make (F : Mirage_flow.S) = struct
       >>= fun server ->
       match event with
       | None -> nexus t fd server input_buffer (List.append pending_promises [ Lwt_mvar.take t.nexus_mbox ])
+      | Some Awa.Server.Userauth (user, userauth) ->
+        let accept = Auth.verify t.user_db user userauth in
+        (* FIXME: Result.get_ok *)
+        let server, reply =
+          Result.get_ok
+            (if accept then
+               Awa.Server.accept_userauth server userauth
+             else
+               Awa.Server.reject_userauth server userauth)
+        in
```

Comment:
This should always return `Ok _`. It can only return `Error _` when you call it with a wrong state. So as commented elsewhere it should maybe be an exception?!